reconcile/utils/gitlab_api.py

```diff
@@ -288,16 +290,17 @@ def _is_bot_username(username: str) -> bool:
         """
         return GROUP_BOT_NAME_REGEX.match(username) is not None
 
-    def get_group_members(self, group_name: str) -> list[GLGroupMember]:
-        group = self.get_group_if_exists(group_name)
+    def get_group_members(self, group: Group | None) -> list[GLGroupMember]:
         if group is None:
-            logging.error(group_name + " group not found")
+            logging.error("no group provided")
             return []
         else:
             return [
                 {
+                    "id": m.id,
                     "user": m.username,
                     "access_level": self.get_access_level_string(m.access_level),
+                    "state": m.state,
```

Comment:
can we remove `GLGroupMember`, to return `GroupMember` directly, then it can be simpler and save another request in https://github.com/app-sre/qontract-reconcile/pull/4762#discussion_r1843218578

Reply:
Thought about it...But  `GroupMember` obj is derived from `group.member.get`...it cannot be created for a new member to be added to the group...